### ISSUE TITLE
SQL Server ALTER TABLE adding column

### ISSUE THREAD

**Issue by alextech** [zendframework/zend-db#215]
SQL Server `Ddl\AlterTable` `addColumn` produces

```
 ALTER TABLE [users]
 ADD COLUMN [profile_image] INTEGER
```

Should be

```
ALTER TABLE [users]
ADD [profile_image] INTEGER
```

**Comment by froschdesign** [comment]
Add the moment there is no decorator to manipulate the `ALTER TABLE` syntax from the SQL Server.

**Comment by alextech** [comment]
Yes, I will add it the same way I added decorator for PostgreSQL to fix index bugs. Just felt sad that a platform which is in list of supported ones gives this many syntax errors on basic commands so adding this to my todo list after I finish fixing sequence bugs.

**Comment by froschdesign** [comment]
> Just felt sad that a platform which is in list of supported ones...

Where?

**Comment by alextech** [comment]
@froschdesign 
https://docs.zendframework.com/zend-db/adapter/

>The list of officially supported drivers:
IbmDb2: The ext/ibm_db2 deriver
Mysqli: The ext/mysqli driver
Oci8: The ext/oci8 driver
Pgsql: The ext/pgsql driver
Sqlsrv: The ext/sqlsrv driver (from Microsoft)
Pdo_Mysql: MySQL via the PDO extension
Pdo_Sqlite: SQLite via the PDO extension
Pdo_Pgsql: PostgreSQL via the PDO extension

**Comment by froschdesign** [comment]
@alextech 
Adapter not DDL Abstraction!

> The following platforms have platform specializations for DDL:
>
> * MySQL
> * All databases compatible with ANSI SQL92


https://docs.zendframework.com/zend-db/sql-ddl/

**Comment by alextech** [comment]
Support for driver does not imply support across the whole package? That is unexpected, but OK, point taken. In that case, last 3 issues, and probably many more undiscovered ones, do not need fixing that much, can be closed, since according to that other link its not officially supported and effort taken to add full support may not be worth it, unless you disagree.

**Comment by froschdesign** [comment]
The priority is not very high for the DDL Abstraction. (imo)

**Comment by alextech** [comment]
That is fine.

**Comment by ezimuel** [comment]
@froschdesign we are already supporting DDL for specific platform like SQL Server, see [here](https://github.com/zendframework/zend-db/tree/master/src/Sql/Platform/SqlServer/Ddl). 
The documentation should be updated https://docs.zendframework.com/zend-db/sql-ddl/.
That said, I would love to see a PR from @alextech with a decorator for AlterTable on SqlServer.

**Comment by alextech** [comment]
@ezimuel I am on it! Thank you for motivation. Even though I was mostly focused on sequences, not being able to set up migrations to support testing of higher-level featured due to these API limitations was a significant annoyance so this distraction will help.

Question about workflow. You assigned it to yourself so does that mean you are working on it or you are willing to review what I will submit? I do not want to cause extra difficulties for you submitting conflicting PRs.

**Comment by ezimuel** [comment]
@alextech I assigned to myself so I remember to manage it. Please, work on the PR and I'll review it, and thanks in advance for your contribution!

**Comment by froschdesign** [comment]
@ezimuel 
> we are already supporting DDL for specific platform like SQL Server

Right, but not for the `ALTER TABLE` syntax. 😉

**Comment by ezimuel** [comment]
@froschdesign right, but we can add it :)

**Comment by alextech** [comment]
yes because we already have high profile public developers saying things like https://twitter.com/DragonBe/status/832182873977413632
because they saw PR for vagrant and such. So it looks like the whole thing is ready. Will try to run it, get nice crashes, and then say bad words.

**Comment by michalbundyra** [comment]
This repository has been closed and moved to laminas/laminas-db; a new issue has been opened at https://github.com/laminas/laminas-db/issues/82.
